src/Network/Ethereum/Core/BigNumber.purs

```diff
@@ -119,13 +121,25 @@ unsafeToInt = Int.floor <<< toNumber
 -- | Take the integer part of a big number
 foreign import floorBigNumber :: BigNumber -> BigNumber
 
-foreign import toBigNumber :: Foreign -> BigNumber
-
 instance decodeBigNumber :: Decode BigNumber where
-  decode = pure <<< toBigNumber
+  decode a = do
+    str <- readString a
+    case parseBigNumber Int.hexadecimal str of
+      Nothing -> fail <<< ForeignError $ "Could not parse BigNumber from stringified Integer: " <> str
+      Just res -> pure res
 
 instance readFBigNumber :: ReadForeign BigNumber where
   readImpl = decode
 
 instance encodeBigNumber :: Encode BigNumber where
   encode = encode <<< (append "0x") <<< toString Int.hexadecimal
+
+instance decodeJsonBigNumber :: A.DecodeJson BigNumber where
+  decodeJson json = do
+    str <- A.decodeJson json
+    case parseBigNumber Int.hexadecimal str of
+      Nothing -> Left $ "Failed to parse BigNumber from hexdecimal string: " <> str
+      Just res -> pure res
+
+instance encodeJsonBigNumber :: A.EncodeJson BigNumber where
+  encodeJson = A.encodeJson <<< (append "0x") <<< toString Int.hexadecimal
```

Comment:
like decode_ we can add `encode_ :: BigNumber -> String` so we have less duplication and less code which can go out of sync